src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -777,7 +777,7 @@ private static string ExtractDeconstructResultElementName(BoundExpression expres
                     }
                 case SyntaxKind.DiscardDesignation:
                     {
-                        var discarded = (DiscardDesignationSyntax)node;
+                        _ = (DiscardDesignationSyntax)node;
```

Comment:
seems like something we'd want to remove.  but that could unfortunately change semantics.  specifically if you wanted to crash on the cast or something.  But that seems super esoteric.  i wonder if we can define a set of operations that should be considered 'safe' and candidates for just total removal.

Reply:
Yes, for now I'd leave this as is and file a separate issue to optimize the analyzer to identify more expressions as those having no side effects.

Reply:
👍

Reply:
Actually, I went ahead and fixed this case in https://github.com/dotnet/roslyn/pull/30777/commits/7140e46c112497a735e291571e9f2932718456d1.